Fix inappropriate roles menu and menuitem in footer links

Stef-Rousset opened the pull request:
#### :tophat: What? Why?
This PR removes inappropriate roles menu and menuitem in footer.
It is issued from the audit of Angers city (page 33), and corresponds to criterias 2.5.3 and 4.1.2 from WCAG.

#### Testing
As a user, go to the dev tools and check that roles menu and menuitem are not present in footer links.

### :camera: Screenshots
<img width="948" alt="456488617-c09273a4-2fed-4508-8af5-31f35876498a" src="https://github.com/user-attachments/assets/7073ba4d-d09a-4332-a867-0fa63fa7a202" />


:hearts: Thank you!


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

* **Bug Fixes**
  * Adjusted menu rendering structure for improved consistency in footer and menu item display

* **Tests**
  * Updated test configurations for election management system

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

alecslupu commented:
@Stef-Rousset Can you fix this pipeline ?

alecslupu commented on decidim-core/app/views/layouts/decidim/shared/_layout_user_profile.html.erb:
```diff
@@ -29,3 +29,17 @@
     </section>
   </div>
 <% end %>
+<script>
```
@Stef-Rousset , can you point where is this being referenced in the provided Audit reports? 

Looking using the WGAC tool provided by Decidim, the user menu is having issues: 
<img width="826" height="450" alt="image" src="https://github.com/user-attachments/assets/4a0c9969-7cdb-4e30-a880-48992060981c" />

Stef-Rousset replied:
hello @alecslupu , thanks for your review ! I made a mistake, this js code wasn't related to this PR, I deleted it.

alecslupu replied:
Apparently there is no mistake with this codebase. Seems that the code path is failing now ... but the issue, seems related to the one in #14951.

alecslupu reviewed (changes requested):
Could you remove this JS, or clarify in which accessibility report should i look into ?

alecslupu reviewed (dismissed):
:+1:

alecslupu commented:
@Stef-Rousset  Any updates on this matter ? ( the same described in #14951)

Stef-Rousset commented:
Hello @alecslupu , I have added the js again for account page, everything seems ok now !

alecslupu commented:
@Stef-Rousset I have checked again this, and the previous error i have mentioned (WGAC error ) can be fixed if i add a timeout on axe script. 

However, if i apply this fix, the user menu will not be collapsed.

Stef-Rousset commented:
> @Stef-Rousset I have checked again this, and the previous error i have mentioned (WGAC error ) can be fixed if i add a timeout on axe script.
> 
> However, if i apply this fix, the user menu will not be collapsed.

Hello @alecslupu , I removed the js added, so that you can apply your fix ! I have also fixed 2 failing tests. However, some tests seems to be unstable, as you can see on the img, the failing tests before I triggered the CI are not the same as the ones that fails now...
 
<img width="946" height="462" alt="Capture d’écran 2025-09-02 à 08 59 06" src="https://github.com/user-attachments/assets/2556a704-eb1c-400d-b77a-2b63c44d5778" />

alecslupu commented:
Having 0.31 releases, we do not longer release 0.29.x series as per policy.  Removing `release: v0.29`

greenwoodt commented:
Hey @Stef-Rousset is this still being worked on? I've noticed there are conflicts and its currently 190 commits behind the `develop` branch. 

Do you have time to take a look into this?

andreslucena commented:
Thanks for your effort with this PR @Stef-Rousset . It seems like it stagnated. @Stef-Rousset  @alecslupu can we retake this please?

andreslucena commented:
@Stef-Rousset can you merge with latest develop so we can check out the specs? Thanks!

andreslucena commented:
Hi @Stef-Rousset please let me know if you're planning on retaking this or should we just takeover this. Thanks!
